src/main/java/com/nike/cerberus/archaius/client/ArchaiusCerberusUrlResolver.java

```diff
@@ -18,47 +18,154 @@
 
 import javax.annotation.Nullable;
 
+import com.amazonaws.regions.Regions;
 import okhttp3.HttpUrl;
 
 import org.apache.commons.configuration.AbstractConfiguration;
-import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.netflix.config.ConfigurationManager;
-import com.nike.cerberus.client.UrlResolver;
-
 /**
  * Class for resolving the Cerberus URL via Archaius.
  */
-public class ArchaiusCerberusUrlResolver implements UrlResolver {
+public class ArchaiusCerberusUrlResolver {
+    public static final String CERBERUS_REGION_ENV_PROPERTY = "CERBERUS_REGION";
+
+    public static final String CERBERUS_REGION_SYS_PROPERTY = "cerberus.region";
 
     public static final String CERBERUS_ADDR_ENV_PROPERTY = "CERBERUS_ADDR";
 
     public static final String CERBERUS_ADDR_SYS_PROPERTY = "cerberus.addr";
 
+    private static final String INVALID_PROPERY_VALUE_TEMPLATE = "Could not find a valid value from the property %s";
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
+    /**
+     * Get a valid region name or null given a possible region name
+     *
+     * @param maybeRegionName
+     * @return null or maybeRegionName
+     */
+    @Nullable
+    private String getValidRegionName(String maybeRegionName) {
+        try {
+            Regions.fromName(maybeRegionName);
```

Comment:
Don't you need to capture the return value from Regions.fromName() and return that?

Reply:
That method returns a Region object instead of a String. It does feel kind of bad to make that class throw an exception to validate the name.